modules/advenueBidAdapter.js

```diff
@@ -0,0 +1,115 @@
+import { registerBidder } from 'src/adapters/bidderFactory';
+import { BANNER, NATIVE, VIDEO } from 'src/mediaTypes';
+import * as utils from 'src/utils';
+
+const BIDDER_CODE = 'advenue';
+const URL = '//supply.advenuemedia.co.uk/?c=o&m=multi';
+const URL_SYNC = '//supply.advenuemedia.co.uk/?c=o&m=cookie';
+
+function isBidResponseValid(bid) {
+  if (!bid.requestId || !bid.cpm || !bid.creativeId || !bid.ttl || !bid.currency) return false;
+  switch (bid.mediaType) {
+    case BANNER:
+      return Boolean(bid.width && bid.height && bid.ad);
+    case VIDEO:
+      return Boolean(bid.vastUrl);
+    case NATIVE:
+      return Boolean(bid.native);
+    default:
+      return false;
+  }
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER, VIDEO, NATIVE],
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {object} bid The bid to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: (bid) => {
+    return Boolean(bid.bidId &&
+        bid.params &&
+        !isNaN(bid.params.placementId) &&
+        spec.supportedMediaTypes.indexOf(bid.params.traffic) !== -1
+    );
+  },
+
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {BidRequest[]} validBidRequests A non-empty list of valid bid requests that should be sent to the Server.
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: (validBidRequests) => {
+    let winTop;
+    try {
+      winTop = utils.getWindowTop();
+      winTop.location.toString();
+    } catch (e) {
+      utils.logMessage(e);
+      winTop = window;
+    };
+
+    const location = utils.getTopWindowLocation();
```

Comment:
Use http://prebid.org/dev-docs/bidder-adaptor.html#referrers
These functions will be removed soon

Reply:
replaced